### ISSUE TITLE
GitLab Group(s) option on file based config is space seperated, not comma

### ISSUE THREAD

**Issue by linuxgemini** [oauth2-proxy/oauth2-proxy#637]
The option `gitlab_group` on the cfg file works as space seperated, not comma seperated as the docs define.

## Expected Behavior

```ini
gitlab_group = "group1,group2"
```

https://github.com/oauth2-proxy/oauth2-proxy/blob/3686b0b44269ef5d7f63f471702e424c42ba5497/docs/2_auth.md#L158

https://github.com/oauth2-proxy/oauth2-proxy/blob/3686b0b44269ef5d7f63f471702e424c42ba5497/docs/configuration/configuration.md#L60

## Current Behavior

```ini
gitlab_group = "group1 group2"
```

## Possible Solution

If there are no side-effects or changes beforehand, `strings.Split(p.Group, " ")` can be changed to `strings.Split(p.Group, ",")`

https://github.com/oauth2-proxy/oauth2-proxy/blob/3686b0b44269ef5d7f63f471702e424c42ba5497/providers/gitlab.go#L176

## Your Environment

The system runs `Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-99-generic x86_64)` with systemd version 237.

- Version used: v5.1.1

**Comment by wstarzak** [comment]
Exactly - i have faced this issue yesterday

**Comment by JoelSpeed** [comment]
We are about to do a major release which already carries lots of breaking changes, it would be good to get this into that if we are going to fix it.

I would actually suggest the fix to be to convert the option to match `GoogleGroup` by converting it to a slice of strings, that way the flag parsing package will do the split for us
https://github.com/oauth2-proxy/oauth2-proxy/blob/3686b0b44269ef5d7f63f471702e424c42ba5497/pkg/apis/options/options.go#L53

This line would become:
```
GitLabGroup              []string   `flag:"gitlab-group" cfg:"gitlab_groups"` 
```
And then we would need to update the flag and wherever it is consumed to accept a list instead of a string

**Comment by linuxgemini** [comment]
I think your suggestion is a lot better, thanks.
I am eagerly waiting for the new release :)

**Comment by JoelSpeed** [comment]
Does anyone have time to raise a PR for this?

**Comment by linuxgemini** [comment]
Closed with #639
